.github/workflows/ci-core.yml

```diff
@@ -111,8 +131,7 @@ jobs:
             id: core_fuzz
             os: ubuntu22.04-8cores-32GB
     name: Core Tests (${{ matrix.type.cmd }})
-    # We don't directly merge dependabot PRs, so let's not waste the resources
-    if: github.actor != 'dependabot[bot]'
+    if: github.actor != 'dependabot[bot]' && ((github.event_name != 'pull_request' && github.event_name != 'merge_group') || needs.filter.outputs.should-run == 'true')
```

Comment:
It looks like we are repeating a lot of complex boolean expressions in this file. Can we consolidate?